ITEvents/Screens/FullEventView/FullEventViewController.swift

```diff
@@ -9,40 +10,75 @@ class FullEventViewController: UIViewController {
         return stackView
     }()
     
-    private let eventDataService: IEventDataService = EventDataServiceMockImpl()
+    private let similarEventsService: ISimilarEventsDataService = SimilarEventsDataServiceMockImpl()
     var event: Event!
+    private var similarEvents: [Event]?
     
     override func viewDidLoad() {
         super.viewDidLoad()
         let eventView = EventView.initiateAndSetup(with: event)
+        
+        if let source = event.source, let url = event.url {
+            let showEventSource = ShowEventSourceView.initiateAndSetup(with: source, and: url)
+            showEventSource.showSourceButton.addTarget(self, action: #selector(openEventSource), for: .touchUpInside)
```

Comment:
// here
`showEventSource.sourceOpenAction = openEventSource`

// class ShowEventSource
```
var sourceOpenAction: () -> void ?
onButtonClick() {
   if let action = sourceOpenAction {
        action()
   }
}
```

Reply:
and actually it should be moved from here